src/utils.js

```diff
@@ -127,60 +127,57 @@ exports.calcCreepCost = function(body) {
 };
 
 exports.checkConstructionSite = function(objects, structureType, x, y) {
-
-    var borderTiles;
-    if(structureType != 'road' && structureType != 'container' && (x == 1 || x == 48 || y == 1 || y == 48)) {
-        if(x == 1) borderTiles = [[0,y-1],[0,y],[0,y+1]];
-        if(x == 48) borderTiles = [[49,y-1],[49,y],[49,y+1]];
-        if(y == 1) borderTiles = [[x-1,0],[x,0],[x+1,0]];
-        if(y == 48) borderTiles = [[x-1,49],[x,49],[x+1,49]];
```

Comment:
There was a bug for positions near corners (e.g. `x=1`, `y=1`) not considering all borders.

It would have been allowed to build at `X` near the following border (`W` = Wall, `E` = Exit):
```
WWWW
EX..
W...
W...
```